### PR TITLE
parser: correct mistake on maping between tinyint and boolean.

### PR DESCRIPTION
**Pull request by zhexuany**
For ddl query like `create table t (c1 boolean)`
`show create table t` gives different result compared against with mysql. 
This patch will fix this issue.

### CODE REVIEW

**Comment by zhexuany** [comment]
@coocood @jackysp PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -5421,9 +5421,18 @@ Type:
 NumericType:
 	IntegerType OptFieldLen FieldOpts
 	{
+		// For boolean and bool, mysql maps them to tinyint(1)
+		// In tidb, tinyint has 4 bits by default. This if branch is to
+		// handle this situation.
+		var x *types.FieldType
+		if b, ok :=  $1.(*types.FieldType); ok {
+			x = b
+			x.Flen = 1
```
Comment (coocood) [inline parser/parser.y]: The `Flen` should be set in `Bool` or `Boolean` entry.

Comment (zhexuany) [reply parser/parser.y]: @coocood PTAL again. I just change to a new to fix this bug.

Comment (coocood) [reply parser/parser.y]: And the return type of an item should be consistent.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Comment by zhexuany** [comment]
@jackysp It's ready to merge now.

**Pull request merged**